pkg/agent/openflow/feature.go

```diff
@@ -0,0 +1,280 @@
+	pipeline binding.PipelineID
+	ofTable  binding.Table
+}
+
+// tableOrderCache is used to save the order of all defined tables located in file pkg/agent/openflow/pipeline.go. The tables
+// have the same pipeline ID are saved in a slice by order of definition. When building a pipeline, the required table list
+// is aggregated from all activated features, and the order of the required tables is decided by the saved order.
+var tableOrderCache map[binding.PipelineID][]*Table
+
+// newTable is used to declare a Table. A table should belong to a stage and pipeline defined in current file.
+func newTable(tableName string, stage binding.StageID, pipeline binding.PipelineID) *Table {
+	table := &Table{
+		name:     tableName,
+		stage:    stage,
+		pipeline: pipeline,
+	}
+	if pipeline == pipelineRoot {
+		tableOrderCache = make(map[binding.PipelineID][]*Table)
+	} else {
+		tableOrderCache[pipeline] = append(tableOrderCache[pipeline], table)
+	}
+	return table
+}
+
+func (t *Table) GetID() uint8 {
+	return t.ofTable.GetID()
+}
+
+func (t *Table) GetNext() uint8 {
+	return t.ofTable.GetNext()
+}
+
+func (t *Table) GetName() string {
+	return t.name
+}
+
+func (t *Table) GetMissAction() binding.MissActionType {
+	return t.ofTable.GetMissAction()
+}
+
+func (f *featurePodConnectivity) getRequiredTables() []*Table {
+	tables := []*Table{
+		// pipelineIP, stageClassifier
+		ClassifierTable,
+		// pipelineIP, stageValidation
+		SpoofGuardTable,
+		// pipelineIP, stageConntrackState
+		ConntrackTable,
+		ConntrackStateTable,
+		// pipelineIP, stageRouting
+		L3ForwardingTable,
+		L3DecTTLTable,
+		// pipelineIP, stageSwitching
+		L2ForwardingCalcTable,
+		// pipelineIP, stageConntrack
+		ConntrackCommitTable,
+		// pipelineIP, stageOutput
+		L2ForwardingOutTable,
+	}
+
+	for _, ipProtocol := range f.ipProtocols {
+		switch ipProtocol {
+		case binding.ProtocolIPv6:
+			tables = append(tables, IPv6Table) // pipelineIP, stageValidation
+		case binding.ProtocolIP:
+			tables = append(tables,
+				// pipelineARP, stageValidation
+				ARPSpoofGuardTable,
+				// pipelineARP, stageOutput
+				ARPResponderTable)
+			if f.enableMulticast {
+				// pipelineIP, stageValidation
+				tables = append(tables, IPClassifierTable)
+			}
+		}
+	}
+
+	return tables
+}
+
+func (f *featureNetworkPolicy) getRequiredTables() []*Table {
+	tables := []*Table{
+		// pipelineIP, stageEgressSecurity
+		EgressRuleTable,
+		EgressDefaultTable,
+		EgressMetricTable,
+		// pipelineIP, stageIngressSecurity
+		IngressSecurityClassifierTable,
+		IngressRuleTable,
+		IngressDefaultTable,
+		IngressMetricTable,
+	}
+	if f.enableAntreaPolicy {
+		tables = append(tables,
+			// pipelineIP, stageEgressSecurity
+			AntreaPolicyEgressRuleTable,
+			// pipelineIP, stageIngressSecurity
+			AntreaPolicyIngressRuleTable,
+		)
+	}
+
+	return tables
+}
+
+func (f *featureService) getRequiredTables() []*Table {
+	if f.enableProxy {
```

Comment:
if it handles `!f.enableProxy` first, the major code don't have to be indented

Reply:
Agreed.